api/app/src/command/medical/facility/update-facility.ts

```diff
@@ -16,9 +16,8 @@ export const updateFacility = async ({
     },
     { where: { id, cxId }, returning: true }
   );
-  if (count != 1)
-    throw new BadRequestError(
-      `Expected a single facility to be updated, but ${count} were updated for id: ${id} and cxId: ${cxId}`
-    );
+  if (count < 1) throw new NotFoundError();
+  // TODO #156 Send this to Sentry
+  if (count > 1) console.error(`Updated ${count} facilities for id ${id} and cxId ${cxId}`);
```

Comment:
With the previous code review comments, I thought the intention was to throw separate errors if the object isn't found vs more than one being updated.

With all of these update functions updating more than one entity should theoretically never happen, and if it does, it's a data integrity and/or programming error. In this case shouldn't we throw so that the incorrect data isn't able to further be modified?

Reply:
The data will be updated anyways, and throwing an error leaves the client wondering whether the changes were applied or not.

Also, the client might attempt to retry the operation, exacerbating the issue.

Throwing also doesn't solve the issue today, as those are quietly taken to logs and we don't get to know about them - https://github.com/metriport/metriport-internal/issues/156 FTW!

Like what you said, it shouldn't happen; if it does, we should be warned asap - which is what #156 will do. I don't see throwing an error helping here.

What do you think?

Reply:
Good point regarding the fact the data will be updated anyways and the cx will probably attempt to retry.

Makes sense to send this to some place where we can be notified in that case.